mailpoet/lib/Segments/DynamicSegments/SegmentSaveController.php

```diff
@@ -38,6 +38,11 @@ public function __construct(
   public function save(array $data = []): SegmentEntity {
     $id = isset($data['id']) ? (int)$data['id'] : null;
     $name = $data['name'] ?? '';
+
+    if (!$this->segmentsRepository->isNameUnique($name, null) && isset($data['force_creation']) && $data['force_creation'] === 'true') {
+      $name = $name . ' (' . wp_generate_password(12, false) . ')';
```

Comment:
Since it's only for the name, I guess we could consider making the random hash shorter...

Reply:
I went with 12 just because it is the default value for `wp_generate_password()`. Making it shorter sounds good to me. Do you have a suggestion of what length to use?

Reply:
@rodrigoprimo Anything that "looks fine" in the listing, I guess. Like 5-6, maybe?

Reply:
I changed it to 5